src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -80,11 +80,11 @@ void AdaptDispatch::_WriteToBuffer(const std::wstring_view string)
     // Turn off the cursor until we're done, so it isn't refreshed unnecessarily.
     cursor.SetIsOn(false);
 
-    // The width at which we wrap is determined by the line rendition attribute.
-    auto lineWidth = textBuffer.GetLineWidth(cursorPosition.y);
+    RowWriteState state{
+        .text = string,
+    };
```

Comment:
I'm afraid this last change is not going to work for me. Because the `AdaptDispatch` class needs to be able to adjust that line width based on the horizontal margins. Storing it in the state should still be OK I think, as long as the initialization of `columnLimit` can be done somewhere here rather than in `TextBuffer::WriteLine`.

Reply:
Hmm I see... And I believe I have to modify
```cpp
r.SetWrapForced(wrapAtEOL && state.columnEnd >= lineWidth);
```
to
```cpp
if (wrapAtEOL) {
    r.SetWrapForced(state.columnEnd >= lineWidth);
}
```
because you're going to set `wrapAtEOL` to `false` when writing with margins, right?

Reply:
I think it probably only needs to be initialized in the one place though: just before `Mode::InsertReplace` check, so the `delayedEOLWrap` code would have had its chance to move to a new line if needed.

Reply:
> because you're going to set `wrapAtEOL` to `false` when writing with margins, right?

I'm not actually doing anything with the `SetWrapForced` state in my current margin implementation, but I think the best solution is probably to disable all wrapping when horizontal margin mode is enabled. It's a bit like the alt buffer mode in that sense.